test/uk/gov/hmrc/incometaxsoftwarechoicesfrontend/views/SearchSoftwareViewSpec.scala

```diff
@@ -26,13 +26,615 @@ import uk.gov.hmrc.incometaxsoftwarechoicesfrontend.forms.FiltersForm
+                              def firstVendor: Element = softwareVendorsSection.selectHead("#software-vendor-0")
+
+                              "has a heading link for the software vendor" in {
+                                val headingLink: Element = firstVendor.selectHead("h3").selectHead("a")
+                                val firstVendorInModel = SearchSoftwarePageContent.softwareVendorsResults.vendors.head
+                                headingLink.attr("href") shouldBe ProductDetailsController.show(Some(firstVendorInModel.name)).url
+                                headingLink.text shouldBe s"${firstVendorInModel.name}"
+                              }
+                            }
+
+                            "have a line for software vendor two" which {
+                              def secondVendor: Element = softwareVendorsSection.selectHead("#software-vendor-1")
+
+                              "has a heading link for the software vendor" in {
+                                val headingLink: Element = secondVendor.selectHead("h3").selectHead("a")
+                                val secondVendorInModel = SearchSoftwarePageContent.softwareVendorsResults.vendors(1)
+                                headingLink.attr("href") shouldBe ProductDetailsController.show(Some(secondVendorInModel.name)).url
+                                headingLink.text shouldBe s"${secondVendorInModel.name}"
+                              }
+                            }
+                          }
+                        }
+                      } else {
+                        "have a alpha software vendor section" which {
+                          val softwareVendorsSection = SearchSoftwareViewSpec.softwareVendorsSection(document)
+
+                          "has a count of the number of software vendors on the page" in {
+                            softwareVendorsSection.selectHead("h3").text shouldBe SearchSoftwarePageContent.numberOfVendorsAlpha
+                          }
+
+                          "has a search form" which {
+                            "contains a heading" in {
+                              document.mainContent.selectHead("form h2 label").text shouldBe SearchSoftwarePageContent.searchFormHeading
+                            }
+
+                            "contains a text input" in {
+                              document.mainContent.selectHead("#searchTerm").attr("name") shouldBe "searchTerm"
+                              if (!hasError)
+                                document.mainContent.selectHead("#searchTerm").attr("value") shouldBe "search test"
+                            }
+
+                            "contains a submit" in {
+                              document.mainContent.selectHead("#searchButton").text shouldBe SearchSoftwarePageContent.searchFormHeading
+                            }
+
+                            if (hasError) {
+                              "contains an error" in {
```

Comment:
The `"contains an error"` test is missing from `"have a beta software vendor section"` section.